templates/default/matterhorn-init-script.erb

```diff
@@ -59,6 +66,20 @@ start() {
 	# the file (flock does not remove the file automatically):
 	[ ! $retval -eq 0 ] && rm -f $lockfile
 
+  # Attempt to restart the daemon until it actually looks like it's
+  # started correctly
+  sleep 10
+  if [ "$attempts" -lt "$max_start_attempts" ]
+    then
+      if ! $(/usr/bin/curl -s -L http://localhost/ | grep -q "$test_string"); then
+        attempts=$[$attempts + 1]
+        restart
+      fi
```

Comment:
Dropping a log line about the restart would identify start up issues here. It would also identify if the 10 seconds was too soon to be checking.